### PR TITLE
Fixes to code examples in jdoc and spec

### PR DESCRIPTION
**Pull request by gavinking**

### CODE REVIEW

File: api/src/main/java/jakarta/data/page/PageRequest.java [hunk 1]
```diff
@@ -37,17 +37,27 @@
...
+ *         PageRequest.of(Person.class).size(100));
+ * var results = page.content();
  * ...
- * for ({@code PageRequest<Person>} p = PageRequest.of(Person.class).size(100);
- *      p != null; p = page.length == 0 ? null : p.next()) {
- *   page = people.findByAgeBetween(35, 59, p);
+ * while (page.hasNext()) {
+ *     page = people.inAgeRange(35, 59,
+ *             page.nextPageRequest().withoutTotal());
+ *     results = page.content();
```
Comment (gavinking) [inline api/src/main/java/jakarta/data/page/PageRequest.java:60]: 👀 this code didn't even compile

File: api/src/main/java/jakarta/data/page/PageRequest.java [hunk 2]
```diff
@@ -37,17 +37,27 @@
  * {@code List}, {@code Stream}, {@link Page}, or {@link CursoredPage}.
  * The parameter of type {@code PageRequest} must occur after the method
  * parameters representing regular parameters of the query itself. For
- * example,</p>
+ * example:</p>
  *
  * <pre>
+ * &#64;Find
```
Comment (njr-11) [inline api/src/main/java/jakarta/data/page/PageRequest.java:43]: Parameter-based automatic query methods don't currently have the ability to do a Between comparison. We will need to wait until next version to be able to replace this one.
```suggestion
```

File: api/src/main/java/jakarta/data/page/PageRequest.java [hunk 3]
```diff
@@ -37,17 +37,27 @@
...
- * Person[] findByAgeBetween(int minAge, int maxAge, {@code PageRequest<Person>} pageRequest);
+ * Person[] inAgeRange(int minAge, int maxAge,
+ *         PageRequest&lt;Person&gt; pageRequest);
+ * </pre>
+ *
+ * <p>This method might be called as follows:</p>
  *
+ * <pre>
+ * var page = people.inAgeRange(35, 59,
+ *         PageRequest.of(Person.class).size(100));
```
Comment (njr-11) [inline api/src/main/java/jakarta/data/page/PageRequest.java:54]: Needs Query by Method Name for now,
```suggestion
 * Person[] findByAgeBetween(int minAge, int maxAge,
 *         PageRequest&lt;Person&gt; pageRequest);
 * </pre>
 *
 * <p>This method might be called as follows:</p>
 *
 * <pre>
 * var page = people.findByAgeBetween(35, 59,
 *         PageRequest.of(Person.class).size(100));
```

File: api/src/main/java/jakarta/data/page/PageRequest.java [hunk 4]
```diff
@@ -37,17 +37,27 @@
...
+ * var results = page.content();
  * ...
- * for ({@code PageRequest<Person>} p = PageRequest.of(Person.class).size(100);
- *      p != null; p = page.length == 0 ? null : p.next()) {
- *   page = people.findByAgeBetween(35, 59, p);
+ * while (page.hasNext()) {
+ *     page = people.inAgeRange(35, 59,
```
Comment (njr-11) [inline api/src/main/java/jakarta/data/page/PageRequest.java:58]:
```suggestion
 *     page = people.findByAgeBetween(35, 59,
```

**Review by njr-11 — CHANGES_REQUESTED** [review]
Thanks for spotting the bugs in the examples. One of them went too far and used a feature isn't in Jakarta Data yet so I added suggestions to restore that one to Query by Method Name.

File: api/src/main/java/jakarta/data/page/PageRequest.java (hunk 2 above)
Comment (gavinking) [reply api/src/main/java/jakarta/data/page/PageRequest.java:43]: Haha yeah, sure, my bad, I was not focused on what that method was actually doing.

**Comment by gavinking** [comment]
I applied your suggestions.

**Review by njr-11 — APPROVED** [review]

File: spec/src/main/asciidoc/repository.asciidoc [hunk 5]
```diff
@@ -1526,10 +1526,12 @@ Obtaining the next 10 products that cost $50.00 or more:
 [source,java]
 ----
 float priceMidpoint = 50.0f;
-PageRequest<Product> pageRequest = Order.by(_Product.price.asc(), _Product.id.asc())
-                                        .pageSize(10)
-                                        .afterKey(priceMidpoint, 0L);
-CursoredPage<Product> moreExpensive = products.findByNameLike(pattern, pageRequest);
+PageRequest<Product> pageRequest =
```
Comment (otaviojava) [inline spec/src/main/asciidoc/repository.asciidoc:1529]: Why not include the extra space here as you did at:
https://github.com/jakartaee/data/pull/590/files#diff-451947eabf993881db66ea833734022a76dff66ae9b0d26c2417e187fa3932acR1520

**Pull request merged**